static/src/javascripts/projects/commercial/commercial-metrics.ts

```diff
@@ -25,17 +24,20 @@ const init = (): Promise<void> => {
 	const shouldForceMetrics = getSynchronousTestsToRun().some((test) =>
 		testsToForceMetricsFor.map((t) => t.id).includes(test.id),
 	);
-	const pageViewId = window.guardian.ophan.pageViewId;
-	const browserId = config.get('ophan.browserId') as string | undefined;
 
 	if (isDev || shouldForceMetrics || userIsInSamplingGroup) {
-		document.addEventListener('visibilitychange', function () {
-			if (logged) return;
-			logged = sendCommercialMetrics(pageViewId, browserId, isDev);
-		});
+		document.addEventListener('visibilitychange', sendMetrics);
 	}
 
 	return Promise.resolve();
 };
 
-export { init };
+/**
+ * A way to override the metrics sampling.
+ * It is safe to call this method repeatedly, as per [the `EventListener` docs](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#multiple_identical_event_listeners)
+ */
+const captureCommercialMetrics = (): void => {
```

Comment:
This is nice and I like it. I would like something similar on core-web-vitals. @zekehuntergreen What do you think? Something we can call at any point in the page lifecycle. As I understand it vitals are only sent through onpageunload anyway, so if in the time the page has been loaded, if anyone said they care about recording the vitals, it'll record the vitals.